### ISSUE TITLE
I want to force push changes.

### ISSUE THREAD

**Issue by jakeonrails** [desktop/desktop#11951]
I want to force push changes.

**Comment by billygriffin** [comment]
@jakeonrails Thanks for the idea. We're likely exploring a few things in the near future that will necessitate force pushing in the same way that rebasing does today, so we'll take a look at force pushing in the context of those features (like amending, for example). But we don't want to just support an across-the-board force pushing at least for the moment due to the destructive effect it can have. 

In the meantime, I'm curious in what circumstances you're wishing that Desktop included force pushing to help with our research. Thanks again!

**Comment by jakeonrails** [comment]
@billygriffin a few things where this is a normal part of my workflow:

- `git rebase origin/master` to pick up changes from colleagues or to locally resolve conflicts
- Locally interactive rebase
- Amending commits

Reasons for interactive rebase:

- reword commit messages
- delete / reorder commits / merge commits into one

Not my use case:

- squashing my PR before merge (I never do this, even through Github merge button, all of my commits are atomic/important [IMHO])

In a week of 100% IC work I will usually force push to Github half a dozen times or more.

It's not that much effort to do it from the terminal but it'd be nice to do it from the app, as force push is about the only thing I use the terminal for git-wise except for compulsively checking git status every few minutes 😆

**Comment by billygriffin** [comment]
@jakeonrails That's really helpful context, thank you! Amending, rewording commit messages, squashing commits together are all things we're going to be starting to explore very soon, and for anything that you can do in Desktop, we're definitely going to support force-pushing after those operations are completed. At that point, we'll need to thoughtfully evaluate whether it would be a better choice just to enable force-pushing across the board, but do it in a way that hopefully keeps people from getting into too much trouble.

I really appreciate you sharing such a detailed set of things you're looking to do with it. ❤️

**Comment by tqpcharlie** [comment]
I'd just like to come here and say that this would be **really** helpful for me too, as I use Github Desktop regularly, but I'm not too experienced with the command line, so, whenever I have to rebase, I have to start from scratch on my crappy terminal.

**Comment by yannickrondeau** [comment]
Using GitHub desktop daily now, and the only thing missing so far for me is this force push. One thing I noted that relates to one of the use case described by @jakeonrails relates to commits after rebase. 

Today, I had to rebase my current branch onto our `develop` branch. After some conflicts' resolution, I had to make a commit onto the rebase just to make sure our auto-generation process and resulting files were good.

Doing so prevented me from completing my rebase, GitHub was asking to pull/push. Had to go through the command line to manually force push.

**Update**
I just saw that I should have pushed first _(allowed force push)_, then amend my commit, then push again _(allowed force push)_.
